test/retained_topic_map.hpp

```diff
@@ -260,7 +262,8 @@ class retained_topic_map {
     }
 
     // Insert a value at the specified subscription path
-    size_t insert_or_update(MQTT_NS::string_view topic, Value const &value) {
+    template<typename V>
+    size_t insert_or_update(MQTT_NS::string_view topic, V&& value) {
```

Comment:
If you're going to change this to &&, then instances of std::move() need to be changed to std::forward<V>()

Reply:
Request is closed now, many issues were in the PR